src/Parser/DecimalMoneyParser.php

```diff
@@ -46,7 +46,15 @@ public function parse($money, $forceCurrency = null)
             );
         }
 
-        $currency = new Currency($forceCurrency);
+        /*
+         * This conversion is only required whilst currency can be either a string or a
+         * Currency object.
+         */
+        $currency = $forceCurrency;
+        if (!$currency instanceof Currency) {
```

Comment:
I believe we should deprecate passing a string here and remove that in 4.0

So I propose triggering a deprecation notice here:

``` php
@trigger_error('Passing a currency as string is deprecated since 3.1 and will be removed in 4.0. Please pass a '.Currency::class.' instance.', E_USER_DEPRECATED);
```

Reply:
Good idea. I'll add that in.

Reply:
My instinct tells me that is a bad idea. I cannot recall the last time I used `trigger_error`. I rather use the docblock `@deprecated`, which could give visual feedback to developers if their IDE supports it.

Reply:
`trigger_error` is something I haven't used much lately either, but unless I'm mistaken, there'd be no way to indicate that the use of `parse` is only deprecated when $currency is parsed as a string. I'd be concerned that showing a deprecation notice for all calls of `parse`, including those that pass the correct `Currency` parameter, would be more confusing.

Reply:
This is widely used in the PHP ecosystem. Triggered errors will immediately be bypassed, so no BC break.

@jarobe42 you can and should trigger errors within the if block, only for those calls which received a string, not a currency.